contracts/strategies/solidLizard/SolidLizardStrategy.sol

```diff
@@ -0,0 +1,318 @@
+    if(IERC20(underlying()).balanceOf(address(this)) > 0) {
+      _enterRewardPool();
+    }
+  }
+
+  /*
+  *   In case there are some issues discovered about the pool or underlying asset
+  *   Governance can exit the pool properly
+  *   The function is only used for emergency to exit the pool
+  */
+  function emergencyExit() public onlyGovernance {
+    _emergencyExitRewardPool();
+    _setPausedInvesting(true);
+  }
+
+  /*
+  *   Resumes the ability to invest into the underlying reward pools
+  */
+  function continueInvesting() public onlyGovernance {
+    _setPausedInvesting(false);
+  }
+
+  function unsalvagableTokens(address token) public view returns (bool) {
+    return (token == rewardToken() || token == underlying());
+  }
+
+  function setRewardLiquidationPath(address _token, ILizardRouter.Route[] memory _route) public onlyGovernance {
+    require(_route[_route.length-1].to == weth, "Path should end with WETH");
+    require(_route[0].from == _token, "Path should start with rewardToken");
+
+    delete reward2WETH[_token];
+    for(uint256 i = 0; i < _route.length; i++) {
+      reward2WETH[_token].push(_route[i]);
+    }
+  }
+
+  function setLpLiquidationPath(address _token, ILizardRouter.Route[] memory _route) public onlyGovernance {
+    require(_route[_route.length-1].to == _token, "Path should end with lp token");
+    require(_route[0].from == weth, "Path should start with WETH");
+
+    delete lpLiquidationPath[_token];
+    for(uint256 i = 0; i < _route.length; i++) {
+      lpLiquidationPath[_token].push(_route[i]);
+    }
+  }
+
+  function addRewardToken(address _token, ILizardRouter.Route[] memory _path2WETH) public onlyGovernance {
+    rewardTokens.push(_token);
+    setRewardLiquidationPath(_token, _path2WETH);
+  }
+
+  function _liquidateReward() internal {
+    if (!sell()) {
+      // Profits can be disabled for possible simplified and rapid exit
+      emit ProfitsNotCollected(sell(), false);
+      return;
+    }
+
+    for(uint256 i = 0; i < rewardTokens.length; i++){
+      address token = rewardTokens[i];
+      uint256 rewardBalance = IERC20(token).balanceOf(address(this));
+
+      if (rewardBalance == 0) {
+        continue;
+      }
+      if (reward2WETH[token].length == 0) {
+        continue;
+      }
+
+      IERC20(token).safeApprove(lizardRouter, 0);
+      IERC20(token).safeApprove(lizardRouter, rewardBalance);
+  
+      ILizardRouter(lizardRouter).swapExactTokensForTokens(
+        rewardBalance,
+        1,
+        reward2WETH[token],
+        address(this),
+        block.timestamp
+      );
+    }
+
+    address _rewardToken = rewardToken();
+    uint256 rewardBalance = IERC20(_rewardToken).balanceOf(address(this));
+    _notifyProfitInRewardToken(_rewardToken, rewardBalance);
+    uint256 remainingRewardBalance = IERC20(_rewardToken).balanceOf(address(this));
+
+    if (remainingRewardBalance == 0) {
+      return;
+    }
+
+    address token0 = ILizardPair(underlying()).token0();
```

Comment:
as `underlying()` is called multiple times, we can save a little bit of gas by calling it only once and then using it in local memory.

So do `address _underlying = underlying();`, and then use `_underlying` for the rest of the calls in lines 172, 173 and 220.